server/container/guice/memory-guice/src/main/java/org/apache/james/modules/mailbox/MemoryMailboxModule.java

```diff
@@ -78,6 +80,7 @@ protected void configure() {
         bind(MailboxManager.class).to(InMemoryMailboxManager.class);
         bind(MailboxACLResolver.class).to(UnionMailboxACLResolver.class);
         bind(GroupMembershipResolver.class).to(SimpleGroupMembershipResolver.class);
+        bind(new TypeLiteral<SystemMailboxesProvider<InMemoryId>>(){}).to(new TypeLiteral<SystemMailboxesProviderImpl<InMemoryId>>(){});
```

Comment:
OK, I finally managed to put bindings for generic types in a more appropriate guice module (JMAPModule), thanks to Antoine.
Note that I personally wouldn't have used the word "nice" in such a context, but I do understand your point :smiley: